api/apiFunctions.ts

```diff
@@ -495,3 +494,40 @@ export async function addUserToLeague({
     throw new Error('Error getting user document ID', { cause: error });
   }
 }
+
+/**
+ * Grabs the length of entries in each league.
+ * @param props - Props being passed in.
+ * @param props.leagues - All user leagues.
+ * @returns {Promise<number>} - Length of entries.
+ */
+export async function getAllLeagueEntries({
+  leagues,
+}: {
+  leagues: string[];
+}): Promise<{ totalEntries: number; alive: number }[]> {
+  try {
+    const response = await databases.listDocuments(
+      appwriteConfig.databaseId,
+      Collection.ENTRIES,
+      [Query.limit(5000)],
+    );
+    const entries = leagues.map((leagueId) => {
+      const leagueEntries = response.documents.filter(
+        (entry) => entry.league.$id === leagueId,
+      );
+      const aliveEntries = leagueEntries.filter(
+        (aliveEntry) => aliveEntry.eliminated === false,
+      );
+
+      return {
+        totalEntries: leagueEntries.length,
+        alive: aliveEntries.length,
+      };
+    });
+    return entries;
+  } catch (error) {
+    console.error('Error getting league entries:', error);
+    return [];
```

Comment:
Q: Could you please explain why we're returning an empty array here, considering your functions return type is ` Promise<{ totalEntries: number; alive: number }[]>`?